### PR TITLE
enhance OVAL for enable_fips_mode

### PR DESCRIPTION
**Pull request by vojtapolasek**
#### Description:

- add more tests to check /etc/kernel/cmdline and also /boot/loader/entries/*.conf files
- use correct check based on prodtype, inspired by grub2_template
- if we are on S390X architecture, check /boot/loader/entries/*.conf only, there is no grubenv file

#### Rationale:

- fixes https://bugzilla.redhat.com/show_bug.cgi?id=2129100

#### Review Hints:

1. provision X86 and S390X RHEL 8 and RHEL 9 machines
2. ./build_product rhel8 rhel9
3. on each machine run fips-mode-setup --enable && update-crypto-policies --set fips:ospp
4. run oscap xccdf eval --profile ospp --report result.html --rule xccdf_org.ssgproject.content_rule_enable_fips_mode <datastream.xml>
5. inspect the HTML report to see that checks are checking corect things

### CODE REVIEW

File: linux_os/guide/system/software/integrity/fips/enable_fips_mode/oval/shared.xml [hunk 1]
```diff
@@ -7,19 +7,62 @@
...
       </criteria>
       {{% endif %}}
     </criteria>
   </definition>
+
+  <ind:textfilecontent54_test id="test_fips_1_argument_in_boot_loader_entries_conf"
+  comment="Check if argument fips=1 is present in the line starting with 'options ' in /boot/loader/entries/.*.conf"
+  check="all" check_existence="all_exist" version="1">
+    <ind:object object_ref="object_fips_1_argument_in_boot_loader_entries_conf" />
+    <ind:state state_ref="state_fips_1_argument_in_boot_loader_entries_conf" />
```
Comment (marcusburghardt) [inline linux_os/guide/system/software/integrity/fips/enable_fips_mode/oval/shared.xml:41]: `state_fips_1_argument_in_boot_loader_entries_conf` and `state_fips_1_argument_in_etc_kernel_cmdline` have the same definition. Could you use the same state for both tests and reduce duplication?

File: linux_os/guide/system/software/integrity/fips/enable_fips_mode/oval/shared.xml [hunk 2]
```diff
@@ -7,19 +7,62 @@
       <extend_definition comment="Dracut FIPS module is enabled" definition_ref="enable_dracut_fips_module" />
       <extend_definition comment="system cryptography policy is configured" definition_ref="configure_crypto_policy" />
       <criterion comment="check if system crypto policy selection in var_system_crypto_policy in the profile is set to FIPS" test_ref="test_system_crypto_policy_value" />
-      {{% if product in ["ol8"] %}}
-      <criterion comment="check if the kernel boot parameter is configured for FIPS mode"
-      test_ref="test_grubenv_fips_mode" />
-      {{% elif product in ["rhel8"] %}}
+      <criterion comment="Check if argument fips=1 for Linux kernel is present in /etc/kernel/cmdline" test_ref="test_fips_1_argument_in_etc_kernel_cmdline" />
```
Comment (marcusburghardt) [inline linux_os/guide/system/software/integrity/fips/enable_fips_mode/oval/shared.xml:10]: I believe this condition is not applicable for `ol8` and `rhel8`, correct? Should it be moved to the block below?

**Review by marcusburghardt — CHANGES_REQUESTED** [review]
Thanks for the update. It is difficult to test this rule and I believe this is the reason to not have test scenario scripts. I have some comments. Could you take a look on them, please?

File: linux_os/guide/system/software/integrity/fips/enable_fips_mode/oval/shared.xml (hunk 2 above)
Comment (vojtapolasek) [reply linux_os/guide/system/software/integrity/fips/enable_fips_mode/oval/shared.xml:10]: Hello @marcusburghardt  I think it is applicable for RHEL 8, after enabling FIPS mode the file appears there.

Comment (marcusburghardt) [reply linux_os/guide/system/software/integrity/fips/enable_fips_mode/oval/shared.xml:10]: Ok. Thanks for clarifying @vojtapolasek .

File: linux_os/guide/system/software/integrity/fips/enable_fips_mode/oval/shared.xml (hunk 1 above)
Comment (vojtapolasek) [reply linux_os/guide/system/software/integrity/fips/enable_fips_mode/oval/shared.xml]: Good catch, fixed.

**Review by marcusburghardt — APPROVED** [review]
Thanks for updating this rule @vojtapolasek .

We have discussed privately on how to include test scenarios for this rule and concluded it would be possible but would not be simple. So we can merge this PR now to not block the fix and work on test scenarios in a separate PR.

**Pull request merged**